utils/near-stable-hasher/src/lib.rs

```diff
@@ -21,3 +21,19 @@ impl Hasher for StableHasher {
         self.0.write(bytes)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::StableHasher;
+    use std::hash::Hasher;
+
+    /// Make sure the stable hasher never changes
+    #[test]
+    fn test_stable_hasher() {
+        let mut sh = StableHasher::new();
+
+        sh.write(vec![1, 2, 3, 4, 5].as_slice());
```

Comment:
```suggestion
        sh.write(&[1, 2, 3, 4, 5]);
```

More generally, I feel we have a lot of "just allocate temporary vec" in our code base. This doesn't feel very rustic -- Rust APIs are very deliberately designed to be zero copy/zero allocation. I feel it makes sense to be more careful about such easy small perf (and simplicity) gains as a matter of habit.